pkg/cmd/provisioning/ibmcloud/create_secrets.go

```diff
@@ -0,0 +1,182 @@
+package ibmcloud
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	credreqv1 "github.com/openshift/cloud-credential-operator/pkg/apis/cloudcredential/v1"
+	"github.com/openshift/cloud-credential-operator/pkg/cmd/provisioning"
+)
+
+const (
+	secretManifestsTemplate = `apiVersion: v1
+stringData:
+  ibmcloud_api_key: %s
+kind: Secret
+metadata:
+  name: %s
+  namespace: %s
+type: Opaque`
+
+	manifestsDirName = "manifests"
+)
+
+// APIKeyEnvVar is the environment variable name containing an IBM Cloud API key
+const APIKeyEnvVar = "IC_API_KEY"
+
+var (
+	// CreateOpts captures the options that affect creation of the generated
+	// objects.
+	CreateOpts = options{
+		TargetDir: "",
+	}
+)
+
+// NewCreateSecretsCmd implements the "create-secrets" command for the credentials provisioning
+func NewCreateSecretsCmd() *cobra.Command {
+	createSecretsCmd := &cobra.Command{
+		Use:              "create-secrets",
```

Comment:
@BobbyRadford I was discussing this with the team and I was wondering if a name like `create-shared-secrets` would be better.
I'm comparing this to the existing aws support and there is no equivalent `create-secrets` subcommand (today) for AWS (which isn't surprising b/c of lack of support for passthrough-equivalent mode). So this subcommand feels specific to implementing Passthrough mode and the name of the command should reflect that.

Could we rename this to something like `create-shared-secrets` and have the long description make it clear that this subcommand just takes the API key from an env var and uses it to satisfy all the CredentialsRequests?

Reply:
Hey @joelddiaz no problem. Will make that happen.